src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -819,23 +858,17 @@ internal bool ContainsGenericVariables()
             return ContainsGenericVariables(GetRuntimeTypeChecked());
         }
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern bool SatisfiesConstraints(RuntimeType paramType, IntPtr* pTypeContext, int typeContextLength, IntPtr* pMethodContext, int methodContextLength, RuntimeType toType);
+        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "RuntimeTypeHandle_SatisfiesConstraints")]
+        private static partial Interop.BOOL SatisfiesConstraints(QCallTypeHandle paramType, nint pTypeContext, RuntimeMethodHandleInternal pMethodContext, QCallTypeHandle toType);
```

Comment:
Nit: `pTypeContext` can be `QCallTypeHandle`. Using two different styles to pass a type in a signature does not look right.

Reply:
In this case it can't without changing `QCallTypeHandle` to accept `RuntimeType?`, that is why it is `IntPtr`. I'm fine changing it to that if that is the preference.

Reply:
Oh, I see. The constructor is overly restrictive for no reason. The impl already handles the case. Will fix.